pkg/kubelet/cm/container_manager_linux.go

```diff
@@ -860,23 +856,20 @@ func ensureSystemCgroups(rootCgroupPath string, manager *fs.Manager) error {
 
 		// Check if we have moved all the non-kernel PIDs.
 		if len(pids) == 0 {
-			break
+			return nil
```

Comment:
My understanding is we attempt to find and move non-kernel thread multiple times just so that we don't miss any transient pids. This change makes this loop a single pass at best, and so this is a change in existing logic, bot a simple bug fix. @dashpole is this change an informed decision?

Reply:
> This change makes this loop a single pass at best, and so this is a change in existing logic, not a simple bug fix.

This is incorrect.  This case is only hit when there are no non-kernel pids to move.  In the previous implementation, the `break` breaks out of the loop to the return at the end of the function.  In this case, we just return directly rather than break and return.

Reply:
Ah gotcha. Makes sense